File: December - 15/SakthiDharan01_royal_family_seating.java
import java.util.*;

public class SakthiDharan01_royal_family_seating {
    private static class Node {
        int val; Node left, right;
        Node(int v){val=v;}
    }

    private static List<String> tokens(String input){
        return Arrays.asList(input.replace(","," ").trim().split("\\s+"));
    }

    private static Node build(List<String> vals){
        if (vals.isEmpty()) return null;
        List<Node> nodes = new ArrayList<>();
        for (String v : vals){
            if (v.equalsIgnoreCase("null")) nodes.add(null);
            else nodes.add(new Node(Integer.parseInt(v)));
        }
        Queue<Node> kids = new ArrayDeque<>();
        for (int i = 1; i < nodes.size(); i++) kids.add(nodes.get(i));
        for (Node n : nodes){
            if (n != null){
                n.left = kids.poll();
                n.right = kids.poll();
            }
        }
        return nodes.get(0);
    }

    private static boolean isComplete(Node root){
        if (root == null) return true;
        Queue<Node> q = new ArrayDeque<>();
        q.add(root);
        boolean end = false;
        while (!q.isEmpty()){
            Node cur = q.poll();
            if (cur == null){
                end = true;
            } else {
                if (end) return false;
                q.add(cur.left);
                q.add(cur.right);
            }
        }
        return true;
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        while (sc.hasNextLine()) sb.append(sc.nextLine()).append(' ');
        String input = sb.toString().trim();
        if (input.isEmpty()) return;
        List<String> vals = tokens(input);
        Node root = build(vals);
        System.out.println(Boolean.toString(isComplete(root)).toLowerCase());
        sc.close();
    }
}

File: December - 15/SakthiDharan01_royal_family_seating.py
import sys
from collections import deque

def build_tree(values):
    if not values:
        return None
    nodes = [None if v is None else [v, None, None] for v in values]
    kids = deque(nodes[1:])
    for node in nodes:
        if node is not None:
            if kids:
                node[1] = kids.popleft()
            if kids:
                node[2] = kids.popleft()
    return nodes[0]

def is_complete(root):
    if not root:
        return True
    q = deque([root])
    end = False
    while q:
        node = q.popleft()
        if node is None:
            end = True
        else:
            if end:
                return False
            q.append(node[1])
            q.append(node[2])
    return True

def parse_line(line):
    vals = []
    for tok in line.replace(',', ' ').split():
        if tok.lower() == 'null':
            vals.append(None)
        else:
            vals.append(int(tok))
    return vals

def solve():
    data = sys.stdin.read().strip().splitlines()
    if not data:
        return
    values = parse_line(' '.join(data))
    root = build_tree(values)
    print(str(is_complete(root)).lower())

if __name__ == "__main__":
    solve()
